node-graph/gcore/src/uuid.rs

```diff
@@ -66,4 +66,23 @@ mod uuid_generation {
 	}
 }
 
+use dyn_any::DynAny;
+use dyn_any::StaticType;
 pub use uuid_generation::*;
+
+#[repr(transparent)]
+#[derive(Debug, Clone, Copy, Default, PartialEq, Eq, Hash, PartialOrd, Ord, serde::Serialize, serde::Deserialize, specta::Type, DynAny)]
+pub struct NodeId(pub u64);
+
+// TODO: Find and replace all `NodeId(generate_uuid())` with `NodeId::new()`.
```

Comment:
Iirc for tests the id generation is controlled manually so doing a find and replace might break things in a non deterministic manner

Reply:
I'm not sure what this means. I moved the NodeId from document.rs to here, but didn't change any usages.

Reply:
Yeah, you saw too late that this was just a move